casper/src/test/scala/coop/rchain/casper/util/rholang/RuntimeManagerTest.scala

```diff
@@ -31,6 +34,48 @@ class RuntimeManagerTest extends FlatSpec with Matchers {
     result.status.isFailed should be(true)
   }
 
+  it should "capture rholang parsing errors and charge for parsing" in {
+    val badRholang = """ for(@x <- @"x"; @y <- @"y"){ @"xy"!(x + y) | @"x"!(1) | @"y"!("hi") """
+    val deploy     = ProtoUtil.sourceDeployNow(badRholang)
+    val (_, Seq(result)) =
+      runtimeManager
+        .use(mgr => mgr.computeState(mgr.emptyStateHash, deploy :: Nil))
+        .runSyncUnsafe(10.seconds)
+
+    result.status.isFailed should be(true)
+    result.cost.cost shouldEqual (accounting.parsingCost(badRholang).value)
+  }
+
+  it should "charge for parsing and execution" in {
+    val correctRholang = """ for(@x <- @"x"; @y <- @"y"){ @"xy"!(x + y) | @"x"!(1) | @"y"!(2) }"""
+    val deploy         = ProtoUtil.sourceDeployNow(correctRholang)
+
+    val (reductionCost, result) =
+      mkRuntimeAndManager("casper-runtime-manager-test")
+        .use {
+          case (runtime, runtimeManager) =>
+            implicit val rand: Blake2b512Random = Blake2b512Random(
+              DeployData.toByteArray(ProtoUtil.stripDeployData(deploy))
+            )
+            val initialPhlo = Cost(accounting.MAX_VALUE)
+            for {
+              _             <- runtime.reducer.setPhlo(initialPhlo)
+              term          <- Interpreter[Task].buildNormalizedTerm(deploy.term)
+              _             <- runtime.reducer.inj(term)
+              phlosLeft     <- runtime.reducer.phlo
+              reductionCost = initialPhlo - phlosLeft.cost
+              state         <- runtimeManager.computeState(runtimeManager.emptyStateHash, deploy :: Nil)
+              result        = state._2.head
+            } yield ((reductionCost, result))
+        }
+        .runSyncUnsafe(10.seconds)
+
+    val parsingCost = accounting.parsingCost(correctRholang)
+
+    result.status.isFailed should be(false)
+    result.cost.cost shouldEqual ((parsingCost + reductionCost).value)
```

Comment:
Unless runtimeManager.computeState restore's `cost` its proper meaning? (in the interpreter, 'cost' is actually 'phlo left', and it goes down on additional computaiton ;/)

Reply:
Yes, it's a bit messy. I'll try to sort it out in a follow-up PR